libraries/apollo-mpp-utils/src/jvmMain/kotlin/com/apollographql/apollo3/mpp/utils.kt

```diff
@@ -14,7 +14,7 @@ actual fun currentTimeFormatted(): String {
 }
 
 actual fun currentThreadId(): String {
-  return Thread.currentThread().id.toString()
+  return Thread.currentThread().threadId().toString()
```

Comment:
Can we do that? This is a [Java19 only](https://docs.oracle.com/en/java/javase/19/docs/api/java.base/java/lang/Thread.html) API. Is the Kotlin compiler smart enough to replace it? I would actually have expected the compiler to error here 🤔

Reply:
Interestingly, it fails at runtime:

```kotlin
test.ThreadTests[jvm] > cacheIsNotReadFromTheMainThread[jvm] FAILED
    java.lang.NoSuchMethodError: 'long java.lang.Thread.threadId()'
```

That should be caught at compile time. I'll investigate this...

Reply:
Alright, that's another instance of https://youtrack.jetbrains.com/issue/KT-57077/1.8.20-RC-243-shows-Java-19-warnings-even-if-configured-with-Java-17-toolchain, I merged main with Kotlin `1.8.20-RC` and this is correctly caught at compile time

Reply:
Latest IntelliJ EAPs should detect the toolchain and not show the symbol strikedthrough (in the IDE) but for some reason it's not working: https://youtrack.jetbrains.com/issue/KTIJ-24844

I reverted that change, it should compile now

Reply:
Interesting. I was running 19 and without that change the Kotlin compiler failed on a deprecation warning.

Reply:
We've been running the Kotlin `-RC` versions and the point you branched from had a regression in the Kotlin compiler (KT-57077). It's all good now!